SonarQube.Client/Requests/PagedRequestBase.cs

```diff
@@ -60,13 +65,24 @@ public async override Task<TResponseItem[]> InvokeAsync(HttpClient httpClient, C
 
                 Page++;
             }
-            while (pageResult.Value != null &&
-                // Continue paging until we get a partial page of results i.e. fewer than requested.
-                // NB there is a bug here: should be comparing against the request page size, not the
-                // maximum allowed size. See https://github.com/SonarSource/sonarqube-webclient-dotnet/issues/8
-                pageResult.Value.Length >= MaximumPageSize);
+            while (!ReachedMaxPageNumber() &&
```

Comment:
This seems unnecessarily complicated. Why not just check `allResponseItems.Count` against the limit?

Reply:
Oh wow, how did I not see it. Tunnel-vision!